nrf_security/src/legacy/CMakeLists.txt

```diff
@@ -110,19 +112,40 @@ if(CONFIG_MBEDTLS_LEGACY_CRYPTO_C AND CONFIG_OBERON_BACKEND)
       ${NRF_OBERON_MBEDTLS_INCLUDE_PATH}
       ${NRF_OBERON_INCLUDE_PATH}
   )
-elseif(CONFIG_MBEDTLS_LEGACY_CRYPTO_C AND CONFIG_CC3XX_BACKEND)
+endif()
+
+if(TARGET mbedcrypto_cc3xx_imported)
 
   # Link directly with cc3xx platform/mbedcrypto libraries
   target_link_libraries(${mbedcrypto_target} PRIVATE
     mbedcrypto_cc3xx_imported
   )
 
+  target_link_libraries(mbedcrypto_cc3xx_imported
+    INTERFACE
+      mbedcrypto_base
+  )
+
   # Add cc3xx platform/mbedcrypto includes to mbedcrypto library
   target_include_directories(${mbedcrypto_target}
     INTERFACE
       $<TARGET_PROPERTY:mbedcrypto_cc3xx_imported,INTERFACE_INCLUDE_DIRECTORIES>
   )
+endif()
+
+if(TARGET mbedcrypto_cc3xx_imported)
+  # The CC3XX needs funcionality which is provided by the inluded source files in src_crypto_legacy
+  # Link CC3XX with mbedcrypto_target because of link order dependency
+  target_link_libraries(mbedcrypto_cc3xx_imported INTERFACE
+    ${mbedcrypto_target}
+  )
```

Comment:
using `INTERFACE` will not make `mbedcrypto_cc3xx_imported` link to `${mbedcrypto_target}`, it will only populate the link interface, so that any target linking with `mbedcrypto_cc3xx_imported` will link with `${mbedcrypto_target}`.

https://cmake.org/cmake/help/latest/command/target_link_libraries.html#libraries-for-a-target-and-or-its-dependents

So either the comment is wrong, or the code.
Without knowing the intentions, I can't tell which is correct.